pennylane_lightning/core/src/simulators/lightning_qubit/measurements/MeasurementsLQubit.hpp

```diff
@@ -100,64 +101,40 @@ class Measurements final
     probs(const std::vector<std::size_t> &wires,
           [[maybe_unused]] const std::vector<std::size_t> &device_wires = {})
         -> std::vector<PrecisionT> {
-        // Determining index that would sort the vector.
-        // This information is needed later.
-        const auto sorted_ind_wires = Pennylane::Util::sorting_indices(wires);
-
-        // Sorting wires.
-        std::vector<std::size_t> sorted_wires(wires.size());
-        for (size_t pos = 0; pos < wires.size(); pos++) {
-            sorted_wires[pos] = wires[sorted_ind_wires[pos]];
+        const std::size_t n_wires = wires.size();
+        const std::size_t num_qubits = this->_statevector.getNumQubits();
+        bool is_all_wires = n_wires == num_qubits;
+        for (std::size_t k = 0; k < n_wires; k++) {
+            if (!is_all_wires) {
+                break;
+            }
+            is_all_wires = wires[k] == k;
         }
-
-        // If all wires are requested, dispatch to `this->probs()`
-        if (wires.size() == this->_statevector.getNumQubits() &&
-            wires == sorted_wires) {
+        if (is_all_wires) {
             return this->probs();
         }
 
-        // Determining probabilities for the sorted wires.
         const ComplexT *arr_data = this->_statevector.getData();
 
-        std::size_t num_qubits = this->_statevector.getNumQubits();
+        // Templated 1-4 qubit cases
+        PROBS_SPECIAL_CASE(1);
```

Comment:
Does this mean that current optimization doesn't consider 5+ wires cases? If yes, is there a plan to add this support?

Reply:
No, these are conditions for early return.

Reply:
In its current state, the implementation is only faster for 1-4 targets (OMP_NUM=1; as shown in the description). I implemented the cases 1-8 to check that and there is a generic implementation which is only faster for 1 target.